### PR TITLE
NUTCH-2285 Digest Authentication support for Nutch 2.X REST API.

### PR DESCRIPTION
**Pull request by kamaci**

### CODE REVIEW

File: src/java/org/apache/nutch/api/NutchServer.java [hunk 1]
```diff
@@ -108,8 +118,41 @@ public NutchServer() {
     application.setStatusService(new ErrorStatusService());
     childContext.getAttributes().put(NUTCH_SERVER, this);
 
-    // Attach the application.
-    component.getDefaultHost().attach(application);
+    AuthenticationTypeEnum authenticationType = configManager.get(ConfigResource.DEFAULT)
+            .getEnum("restapi.auth", AuthenticationTypeEnum.NONE);
+
+    if (authenticationType == AuthenticationTypeEnum.NONE) {
```
Comment (lewismc) [inline src/java/org/apache/nutch/api/NutchServer.java]: Please change to the more efficient switch notation. Multiple If's are messy and JDK1.8 has better switch support for string input values.

Comment (kamaci) [inline src/java/org/apache/nutch/api/NutchServer.java]: @lewismc I've added switch with enum support.

**Comment by kamaci** [comment]
@lewismc this PR is ready to check for merging.

**Comment by kamaci** [comment]
I'll create a new pull request which does not have conflicts.

**Pull request closed without merging**